### PR TITLE
Return fit results dict in Fit methods

### PR DESCRIPTION
**Pull request by luca-giunti**
**Description**
As additional output, I appended all fit results to the dictionary returned by `Fit.stat_profile()` and `Fit.stat_surface()`.

**Dear reviewer**
Can we do the same for Estimators? It's a bit more problematic there because they don't return dicts...

### CODE REVIEW

File: gammapy/modeling/fit.py [hunk 1]
```diff
@@ -326,7 +326,8 @@ def stat_profile(
         Returns
         -------
         results : dict
-            Dictionary with keys "values" and "stat".
+            Dictionary with keys "values", "stat" and "results". The latter contains an
```
Comment (AtreyeeS) [inline gammapy/modeling/fit.py:329]: Hi @luca-giunti ! I find it a bit uncomfortable to have a `results` key inside a `results` dict. Probably it is better to remake it and explain what it means?

Comment (luca-giunti) [reply gammapy/modeling/fit.py:329]: Ah right. I could rename the dict key to `"fit_results"`. Would it be Ok then?

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @luca-giunti, I just have the comment to add at least one assert on the returned fit results in our tests somewhere...

**Review by adonath — APPROVED** [review]
Thanks @luca-giunti, no further comments from my side.

**Pull request merged**